### DISCUSSION TITLE
How do you use `useDebounceCallback`?

### DISCUSSION THREAD

**Discussion by benze** [mantinedev/mantine#5664] [Q&A]
I'm starting out with Mantine, and ran across the `useDebounceCallback` hook, however I cannot seem to find any documentation or examples that relate to this.  I tried looking through the commit history to see if I could identify the use case that it is trying to address, but can't even find the commit where it first makes its appearance in Mantine.

Given that the callback function is a `()-> void`, I'm not quite sure what the use case is for something like this. 

Can anyone point me to an example or some background what this could be used for?

**Comment by rtivital** [comment]
Mantine does not provide `useDebounceCallback`. It was never a part of the library.

**Reply by benze** [reply]
@rtivital I found it in the callbacks package

https://github.com/mantinedev/mantine/tree/7.5.0/packages/%40mantine/hooks/src/use-debounced-callback

**Reply by rtivital** [reply]
It is internal function that is not meant to be used by end users

**Reply by benze** [reply]
Then what is the recommended approach to only triggering a callback with a debounced value?  For instance, if I only want to call a search function using a debounced value in a text box, how can I control that?  Using `useDebouncedValue` will ensure a debounced value, but how can I "restrict" calls to my search function only when my debounced value changes?

I saw the https://www.npmjs.com/package/use-debounce library, but with all the other support for debouncing in Mantine, is there a better way?

**Reply by rtivital** [reply]
You can subscribe to debounced value in useEffect

**Reply by benze** [reply]
Good idea; I'll try that.  Is there a particular reason why the use-debounced-callback isn't made a public hook?
